homeassistant/components/hydrawise/sensor.py

```diff
@@ -48,30 +78,83 @@ async def async_setup_entry(
     coordinator: HydrawiseDataUpdateCoordinator = hass.data[DOMAIN][
         config_entry.entry_id
     ]
-    async_add_entities(
-        HydrawiseSensor(coordinator, description, controller, zone)
-        for controller in coordinator.data.controllers.values()
-        for zone in controller.zones
-        for description in SENSOR_TYPES
-    )
+    entities: list[HydrawiseSensor] = []
+    for controller in coordinator.data.controllers.values():
+        entities.extend(
+            HydrawiseSensor(coordinator, description, controller, zone=zone)
+            for zone in controller.zones
+            for description in ZONE_SENSORS
+        )
+        entities.extend(
+            HydrawiseSensor(coordinator, description, controller, sensor=sensor)
+            for sensor in controller.sensors
+            for description in FLOW_CONTROLLER_SENSORS
+            if "flow meter" in sensor.model.name.lower()
+        )
+        entities.extend(
+            HydrawiseSensor(
+                coordinator, description, controller, zone=zone, sensor=sensor
+            )
+            for zone in controller.zones
+            for sensor in controller.sensors
+            for description in FLOW_ZONE_SENSORS
+            if "flow meter" in sensor.model.name.lower()
+        )
+    async_add_entities(entities)
 
 
 class HydrawiseSensor(HydrawiseEntity, SensorEntity):
     """A sensor implementation for Hydrawise device."""
 
-    zone: Zone
+    @property
+    def icon(self) -> str | None:
+        """Icon of the entity based on the value."""
+        if self.entity_description.key in FLOW_MEASUREMENT_KEYS and self.state < 0.01:
```

Comment:
why the state < 0.01?

Reply:
Because the Hydrawise cloud API doesn't report 0.0 but something like 0.00000384. Don't ask me why.

Reply:
I made this a bit cleaner by rounding and then comparing.